server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailHeader.scala

```diff
@@ -75,12 +76,35 @@ object GroupedAddressesHeaderValue extends EmailHeaderValue {
   }
 }
 
+object MessageIdsHeaderValue {
+  def from(field: Field): MessageIdsHeaderValue = {
+    val messageIds: List[HeaderMessageId] = format(field.getBody)
+      .split(" ")
+      .flatMap(body => {
+        if(Pattern.compile("^<.*@.*>$").matcher(body).matches()) {
+          scala.Right(body.replaceAll("[<,>]", ""))
+        } else {
+          Left()
+        }
+      }.toOption)
+      .toList
+      .map(HeaderMessageId(_))
+
+      MessageIdsHeaderValue(Option(messageIds).filter(_.nonEmpty))
+  }
+
+  def format(body: String): String = {
```

Comment:
Uneeded `{}`

Reply:
format => unfold(normalizeSpace(...))

I feel that the format wording is hidding what it does.